From f2cc448dc6fb1010b889a7a3531422e8ddc9967c Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Mon, 21 Dec 2020 10:55:47 +0200
Subject: [PATCH] [SW-0]: fit: copy keys in images folder

Currently, keys are stored in common folder and the path with key name
is set in token KEY_DTB using sed command. In order to avoid setting
complete path for key, this patch copies required keys to images folder,
set KEY_DTB to key name and pass images folder location in mkimage
command to sign images.

Change-Id: I2ec5530b7384bce81f73b0297f40b2785b31aadf
---

diff --git a/board/habanalabs/common/post-image-fit.sh b/board/habanalabs/common/post-image-fit.sh
index c5fda87..b4aeb3e 100755
--- a/board/habanalabs/common/post-image-fit.sh
+++ b/board/habanalabs/common/post-image-fit.sh
@@ -67,8 +67,9 @@
 
 # Set the first found public key file name in dtb format inside the its template
 KEYS_DIR=${COMMON_BASE}/key/u-boot/
+KEY_NAME=hl_rsa3072
 
-ITS_SED_PARAMS+="s#@KEY_DTB@#${KEYS_DIR}/hl_rsa3072.dtb#g;"
+ITS_SED_PARAMS+="s/@KEY_DTB@/${KEY_NAME}.dtb/g;"
 
 function generate_dsv_fit
 {
@@ -80,7 +81,18 @@
 	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${DSV_ITS_FILE} ${DSV_ITB_FILE}
 }
 
+function hl_copy_keys
+{
+	FILEPATH=$1
+
+	cp ${KEYS_DIR}/${KEY_NAME}.key ${FILEPATH}
+	cp ${KEYS_DIR}/${KEY_NAME}.dtb ${FILEPATH}
+}
+
 pushd ${BINARIES_DIR}
+
+	hl_copy_keys ${BINARIES_DIR}
+
 	if [ ${HL_TARGET} == "onix" ]; then
 		sed ${ITS_SED_PARAMS} ${SPI_FPGA_ITS_TEMPLATE} > ${SPI_ITS_FILE}
 		${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${SPI_ITS_FILE} -k ${KEYS_DIR} ${SPI_ITB_FILE}
@@ -89,14 +101,14 @@
 	fi
 
 	sed ${ITS_SED_PARAMS} ${UBOOT_ITS_TEMPLATE} > ${UBOOT_ITS_FILE}
-	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${UBOOT_ITS_FILE} -k ${KEYS_DIR} ${UBOOT_ITB_FILE}
+	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${UBOOT_ITS_FILE} -k ${BINARIES_DIR} ${UBOOT_ITB_FILE}
 
 	# LKD like this to be 4B aligned in size
 	dd status=none conv=sync bs=4 if=${UBOOT_ITB_FILE} > ${HL_DEV}-boot-fit.itb
 
 	sed ${ITS_SED_PARAMS} ${LINUX_ITS_TEMPLATE} > ${LINUX_ITS_FILE}
 	cp ${BASE_DIR}/build/linux-custom/arch/arm64/boot/Image .
-	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${LINUX_ITS_FILE} -k ${KEYS_DIR} ${LINUX_ITB_FILE}
+	${HOST_DIR}/usr/bin/mkimage -D "-W no-unit_address_vs_reg" -f ${LINUX_ITS_FILE} -k ${BINARIES_DIR} ${LINUX_ITB_FILE}
 
 	# KMD like this to be 4B aligned in size
 	dd status=none conv=sync bs=4 if=${LINUX_ITB_FILE} > ${HL_DEV}-fit.itb
